### PR TITLE
Bits of RADTTS support

### PR DESCRIPTION
**Pull request by borisfom**
Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

# What does this PR do ?

Those are modifications needed to support RadTTS net - which, for now, cannot be integrated into Nemo.
Adding those support bits to Nemo will make it easier to make internal RADTTS implementation a  Nemo extension, rather than a fork.   

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by titu1994 — APPROVED** [review]
Overall looks correct.

**Comment by titu1994** [comment]
Requires @blisc review before merge

File: nemo/collections/tts/torch/data.py [hunk 1]
```diff
@@ -277,6 +281,9 @@ def __init__(
         self.sup_data_types_set = set(self.sup_data_types)
 
         for data_type in self.sup_data_types:
+            if data_type not in VALID_SUPPLEMENTARY_DATA_TYPES:
+                raise NotImplementedError(f"Current implementation doesn't support {data_type} type.")
+
```
Comment (XuesongYang) [inline nemo/collections/tts/torch/data.py:286]: Please remove this exception. See details in this PR (https://github.com/NVIDIA/NeMo/pull/4297).

Comment (borisfom) [reply nemo/collections/tts/torch/data.py]: Done.

File: nemo/collections/tts/torch/data.py [hunk 2]
```diff
@@ -450,6 +471,7 @@ def __getitem__(self, index):
         # Load alignment prior matrix if needed
         align_prior_matrix = None
         if AlignPriorMatrix in self.sup_data_types_set:
+            align_prior_matrix = None
```
Comment (XuesongYang) [inline nemo/collections/tts/torch/data.py:474]: pls remove this line because it is redundant.

File: nemo/collections/tts/torch/data.py [hunk 3]
```diff
@@ -37,21 +37,23 @@
 from nemo.collections.tts.torch.tts_data_types import (
     DATA_STR2DATA_CLASS,
     MAIN_DATA_TYPES,
+    VALID_SUPPLEMENTARY_DATA_TYPES,
```
Comment (XuesongYang) [inline nemo/collections/tts/torch/data.py:40]: pls remove this line since it is not used anymore.

**Review by XuesongYang — CHANGES_REQUESTED** [review]

File: nemo/collections/tts/torch/data.py [hunk 4]
```diff
@@ -490,6 +512,44 @@ def __getitem__(self, index):
...
+                    audio.numpy(),
+                    fmin=self.pitch_fmin,
+                    fmax=self.pitch_fmax,
+                    frame_length=self.win_length,
+                    sr=self.sample_rate,
+                    fill_na=0.0,
+                )
+                p_voiced = torch.from_numpy(p_voiced).float()
+                torch.save(p_voiced, p_voiced_path)
+
```
Comment (XuesongYang) [inline nemo/collections/tts/torch/data.py:552]: the operations for `voiced_mask` and `p_voiced` are exactly the same. Could you please define a single member function and make a function call here to remove duplicates?

**Review by XuesongYang — CHANGES_REQUESTED** [review]

File: tests/collections/tts/test_torch_tts.py [hunk 5]
```diff
@@ -23,7 +23,7 @@
 
 
 class TestTTSDataset:
-    @pytest.mark.run_only_on('CPU')
+    # @pytest.mark.run_only_on('CPU')
```
[the remark this answers is not in the document]
Comment (borisfom) [reply tests/collections/tts/test_torch_tts.py:26]: I don't think it works as intended. When I ran unit test locally on my machine, it was skipped, even though it had CPU, obviously. And when I commented it out, it passed.

File: nemo/collections/tts/torch/data.py (hunk 4 above)
Comment (borisfom) [reply nemo/collections/tts/torch/data.py]: Done.

File: tests/collections/tts/test_torch_tts.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (XuesongYang) [reply tests/collections/tts/test_torch_tts.py]: sounds good. thanks for your explanation.

File: nemo/collections/tts/torch/data.py (hunk 4 above)
Comment (XuesongYang) [reply nemo/collections/tts/torch/data.py]: thank you.

**Review by XuesongYang — COMMENTED** [review]
Thanks! LGTM. Let's wait for @blisc for the final review.

**Review by XuesongYang — APPROVED** [review]

**Review by titu1994 — APPROVED** [review]

**Pull request merged**